test/execution/evaluation_test.exs

```diff
@@ -0,0 +1,174 @@
+defmodule Wanda.Execution.EvaluationTest do
+  use ExUnit.Case
+
+  alias Wanda.Execution.{
+    AgentCheckResult,
+    CheckResult,
+    Evaluation,
+    ExpectationEvaluation,
+    ExpectationEvaluationError,
+    ExpectationResult,
+    Result
+  }
+
+  describe "evaluation of an execution" do
```

Comment:
Nice work with the tests. I don't want to add any overload here, but would it make sense to cover more scenarios. I only see the "passing" scenario covered XD

Reply:
ehrm we had it in mind but this morning i totally forgot to add them we can have them in a separate PR, adding a task in jira

Reply:
@arbulu89  added moar tests